substrate/frame/support/test/tests/pallet.rs

```diff
@@ -221,6 +224,7 @@ pub mod pallet {
 		#[pallet::call_index(0)]
 		#[pallet::weight(Weight::from_parts(*foo as u64, 0))]
 		#[deprecated = "test"]
+		#[allow(deprecated)]
```

Comment:
Here the same.

If the answer is, so that the generated code doesn't generate warnings. It should be changed to check for `deprecated` attribute and handle this.

Reply:
The generated code basically should not output any "deprecated" warning. This makes no sense. The deprecating warning should only be visible for "outside" users. 

So, we could check for `deprecated` in the attributes and then put `allow(deprecated)` above the generated code.

Reply:
> The generated code basically should not output any "deprecated" warning. This makes no sense. The deprecating warning should only be visible for "outside" users.
> 
> So, we could check for `deprecated` in the attributes and then put `allow(deprecated)` above the generated code.

And it should be applied to generated code of every kind of item  that has deprecated attribute?

Reply:
`allow(deprecated)` is no longer needed now, done as you suggested for generated code.